substrate/client/network/sync/src/justification_requests.rs

```diff
@@ -146,6 +211,10 @@ impl<B: BlockT> ExtraRequests<B> {
 			}
 			self.failed_requests.entry(request).or_default().push((who, Instant::now()));
 			self.pending_requests.push_front(request);
+			if let Some(metrics) = &self.metrics {
+				metrics.failed.set(self.failed_requests.len().try_into().unwrap_or(u64::MAX));
+				metrics.pending.inc();
+			}
 		} else {
 			trace!(target: LOG_TARGET,
```

Comment:
This is out of scope of this PR, but it looks like we can legitimately hit this if the response was delivered after the peer disconnected event arrived. This can happen depending on the order different protocols/streams are polled as these events are emitted by different objects (notifications protocol for sync peers and request-response protocol for responses).

Created an issue fr this: https://github.com/paritytech/polkadot-sdk/issues/5414.